curriculum/challenges/english/25-front-end-development/lab-real-time-counter/66bb3e20d3dc5b6d0a21f5dd.md

```diff
@@ -0,0 +1,190 @@
+
+**User Stories:**
+
+1. You should have an `h1` element with the text `Real-Time Character Counter`.
+2. You should have a `textarea` element with the `id` of `text-input`.
+3. There should be a `p` element with the `id` of `char-count`. It should initially contain the text `Character Count: 0/50`. This placeholder text should be replaced with the current count of characters in the `textarea` element.
+
+4. When the `#text-input` element contains the text `hello coder` the `#char-count` element should contain the text `"Character Count: 11/50"`
+5. When the `#text-input` element contains the text `i am a programmer` the `#char-count` element should contain the text `"Character Count: 17/50"`
+6. When the `#text-input` element contains the text `hello world` the `#char-count` element should contain the text `"Character Count: 11/50"`
+7. When the `#text-input` element contains the text `I am learning a new language and it is called c++` the `#char-count` element should contain the text `"Character Count: 50/50"`
+
+<!-- 8.  You should attach an event listener to the `text-input` element that listens for the `input` event. 
+9.  Inside the event trigger function, you should get the value of the `textarea` element and store it in the variable `textInput`. You should also declare a variable `charCount` and set it equal to the length of the `textInput` variable.
+10. Create a condition that checks if the current character count is greater than `50`. If it is, the user shouldn't be able to enter more characters. 
+11. Get the character count display element by its ID and update the text to show the current character count using `innerHTML`.
+12. If the character count is equal to `50`, the text should be displayed in `red`.  -->
+
+**Note:** Be sure to link your stylesheet and the JavaScript file in your HTML.
+
+# --hints--
+
+There should be an `h1` element with the text of your topic.
+
+```js
+assert.isAbove(document.querySelector('h1')?.innerText.length, 0);
+```
+
+You should have a `textarea` element with the `id` of `text-input`. 
+
+```js
+assert($('textarea').attr('id') == 'text-input');
+```
+
+You should have a `p` element with the `id` of `char-count`. 
+
+```js
+assert($('p').attr('id') == 'char-count');
+```
+
+The `#char-count` `p` element should initially contain the text `Character Count: 0/50`.
+
+```js
+assert(document.querySelector('p')?.innerText, 'Character Count: 0/50');
+```
+
+When the `#text-input` element contains the text `hello coder` the `#char-count` element should contain the text `"Character Count: 11/50"`
+
+```js
+const textInput = document.getElementById('text-input');
+const charCount = document.getElementById('char-count');
+
+textInput.dispatchEvent(new Event('change'));
+
+assert.strictEqual(charCount.innerText.trim(), 'Character Count: 11/50');
+console.log("current count" + charCount.innerText);
```

Comment:
```suggestion
const textInput = document.getElementById('text-input');
const charCount = document.getElementById('char-count');

textInput.value = "hello coder";
textInput.dispatchEvent(new Event('input'));
textInput.dispatchEvent(new Event('change'));

assert.strictEqual(charCount.innerText.trim(), 'Character Count: 11/50');
```

We need to fire the "input" event too, which is what the solution code listens for. Also, neither of these will really help if we don't change the *value* of the text input first. :3

Reply:
This test is passing even if nothing is entered in the text area.

Reply:
I was expecting something different, this test works!